src/streaming/MediaPlayer.js

```diff
@@ -270,6 +278,8 @@ MediaPlayer = function (context) {
                     doAutoPlay.call(this);
                 }
             }
+
+            this.adapter.reset();
```

Comment:
@greg80303, let me explain why I put adapter.reset in this method - MediaPlayer now has `getStreamsFromManifest` and `getTracksForTypeFromManifest` methods that use DashAdapter's methods that in turn initialize adapter's `periods` and `adaptations` members. If these methods are called for a manifest A and then MediaPlayer.attachSource is called for a manifest B w/o resetting the adapter it will have incorrect data about periods and adaptations. That is why I reset adapter in resetAndPlay. 

I think we need to move adapter.reset right at the top of the resetAndPlay function Since 'playing' and 'streamControllers'  may not be set in case I described above. Does it make sense?